server.py

```diff
@@ -78,6 +78,40 @@ def list_json():
 	return send_from_directory(app.static_folder, "list.json", max_age=20)
 
 
+@app.route("/server/<string:ip>/<int:port>")
+def get_server(ip, port):
+    try:
+        # Validate IP address
```

Comment:
Do we need this validation? Since it's just a lookup, you will just get a "server not found" if you pass an invalid address. (Furthermore, such validation may itself introduce DoS attack vectors: For example consider the string -> int conversion for arbitrarily large strings below. Arguably Python does limit that however.)

What I'm slightly more concerned about: This seems to not normalize IPv6 addresses; it seems to not allow omitting zeroes from IPv6 addresses (since then `not part` will be `true` below)?

This seems especially problematic as in the serverlist, there are IPv6 addresses with omitted zeroes (example: "Free For All Skywars" has `2a02:4780:28:71f7::1`).

Reply:
These are fair points. I thought about removing the IP validation, but only after I had committed. I can definitely still remove it considering the only advantage is slightly better error messages.

Reply:
Okay, but that still leaves the point about normalization.

I feel like this should be relatively easy to do properly by using Python's built-in `ipaddress` library rather than manually juggling (generally not normalized) IP address strings everywhere.

Reply:
I'm not sure. How verbose and specific do we need error messages to be. Really it either finds a match or it doesn't. Wouldn't this be better implemented in the `serverList.get()` function itself?

Reply:
I'm not concerned about the error messages. I think the serverlist should probably be using IP objects internally to begin with, and this PR should look up things in the server list by IP object equality.

> Really it either finds a match or it doesn't.

The problem is that finding a match by string equality is bogus for IPv6 addresses since the same address can be written in different ways (you can not omit the zeroes, for example).

You could make the formatting part of the API but I don't like that.

Then again of course refactoring the serverlist to use IP address objects may be out of scope.